src/App.tsx

```diff
@@ -26,70 +29,132 @@ type ReorderOptions = {
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
-
-      <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
-      </ul>
-    </div>
-  );
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
 };
+
+export class App extends Component<{}, State> {
+  readonly state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  handleSortAlphabetic = () => {
+    this.setState({
+      sortType: SortType.ALPHABET,
+    });
+  };
+
+  handleSortByLength = () => {
+    this.setState({
+      sortType: SortType.LENGTH,
+    });
+  };
+
+  handleReverse = (prevState: State) => {
+    this.setState({
+      isReversed: !prevState.isReversed,
+    });
+  };
+
+  handleReset = () => {
+    this.setState({
+      sortType: SortType.NONE,
+      isReversed: false,
+    });
+  };
+
+  render() {
+    const {
+      isReversed: isRev,
+      sortType: sortBy,
+    } = this.state;
+
+    const options = {
+      sortType: sortBy,
+      isReversed: isRev,
+    };
+
+    const isReordered = isRev || sortBy !== SortType.NONE;
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={
+              classNames('button', 'is-info',
+                { 'is-light': sortBy !== SortType.ALPHABET })
+            }
+            onClick={() => this.handleSortAlphabetic()}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={
+              classNames('button', 'is-success',
+                { 'is-light': sortBy !== SortType.LENGTH })
+            }
+            onClick={() => this.handleSortByLength()}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={
+              classNames('button', 'is-warning', { 'is-light': !isRev })
+            }
+            onClick={() => this.handleReverse(this.state)}
+          >
+            Reverse
+          </button>
+
+          {isReordered && (
+            <button
+              type="button"
+              className="button is-danger is-light"
+              onClick={() => this.handleReset()}
+            >
+              Reset
+            </button>
+          )}
+        </div>
+
+        <GoodsList goods={getReorderedGoods(
+          goodsFromServer,
+          options,
+        )}
+        />
```

Comment:
```suggestion
        <GoodsList
          goods={getReorderedGoods(
            goodsFromServer,
            options,
          )}
        />
```

Reply:
had been refactored to improve readability
thank you for your feedback